### ISSUE TITLE
Question: How to make open (enter) use the editor for files, but keep its directory behaviour?

### ISSUE THREAD

**Issue by shaypal5** [Canop/broot#481]
Hey,

I'm working on MacOS, and having the `open` system command invoked on the file selecting when I press enter doesn't make a lot of sense, because I don't use the terminal to search for apps or images, only for text, code and configuration files - meaning only things for which I want to stay in the context of the terminal.

However, even If I set Vim as the default app to open `.py` files with, MacOS will still open Vim as a separate app in a separate window; it won't invoke `vim my_file.py` as a command in the same terminal session I'm in. This is imporant to me, since I use `byobu` (basically `tmux`) and work with several tabs, each with several splits. I want to stay in context when I'm in a certain terminal split and want to search for a file.

Anyway, I've managed to make `enter` open any file I'm selecting in `vim` using the following configuration:
```hjson
{
    invocation: openeditleave
    key: enter
    execution: "$EDITOR +{line} {file}"
    leave_broot: true
}
```

But now pressing `enter` on a directory also opens it in `vim`. I know I'm doing something wrong, and I'd love some help. :)

Thank you, 
Shy

**Comment by Canop** [comment]
There's no filtering on file extension yet but you can chose to apply a verb either on directories only or on files only:

```Hjson
{
    invocation: openeditleave
    key: enter
    execution: "$EDITOR +{line} {file}"
    leave_broot: true
    apply_to: file
}
```

See https://dystroy.org/broot/conf_verbs/#verb-definition-attributes

**Comment by shaypal5** [comment]
That works great! Thank you!

Well.... While we're at it, I can't make `alt` differentiate the enter command with-vs-without leaving broot:
```
{
    invocation: openeditleave
    key: enter
    execution: "$EDITOR +{line} {file}"
    leave_broot: true
    apply_to: file
}
{
    invocation: openeditstay
    key: ctrl-enter
    execution: "$EDITOR +{line} {file}"
    leave_broot: false
    apply_to: file
}
```

ctrl-enter isn't bound by anything special I have (like iTerm shortcuts or Karabiner Elements). You have any idea what might be the case?

**Comment by Canop** [comment]
There are a lot of layers here. You already know the system extensions and terminals can intercept a lot of key combinations. If you use some kind of terminal multiplexer, it can intercept some (and break some other ones). Then, there's the protocol used for communication between the terminal and the hosted applications. The most common one prevents a lot of combinations (for outdated dumb reasons, but we can't change it).

There's some work to deal with new APIs for an improved support for combinations (that not all terminals support) but it's not yet here.

In the meantime, you can use broot's log to see whether the shortcut is received by broot: https://dystroy.org/broot/community/#log

**Comment by shaypal5** [comment]
Thanks. I'll work it out from there!

**Comment by shaypal5** [comment]
Just for anyone else that stumbles into this issue:

I set a (what seems to be) standard remap for alt-enter (or option-return, as it's called on mac) for iTerm2:
Going to *Preferences->Profiles->Default->Keys* and mapping `alt-return` to `Send Hex Codes: 0x1b 0x0d`

I found this in several comments on this issue:
https://github.com/Canop/broot/issues/86#issuecomment-801877468

In conjunction with the following custom verb, I got my desired behavior (`enter` opens a file for editing on vim while closing broot, while `alt-enter` leaves broot open, returning to it when I exit vim):

```
{
    invocation: openeditstay
    key: alt-enter
    execution: "$EDITOR +{line} {file}"
    leave_broot: false
    apply_to: file
}
```

**Comment by Canop** [comment]
@shaypal5 It looks like this FAQ entry should be updated: https://dystroy.org/broot/common-problems/#alt-enter-or-other-shortcut-isnt-available

Do you agree ? Do you see what would be the clearest for mac users ?

**Comment by shaypal5** [comment]
Yes. First, I've disabled my custom verb and checked that `alt-enter` yields the default `:open_leave` behavior for me in such a case. So we can be pretty sure that adding just this single mapping on iTerm2 restates the default behavior for the average user.

It has a minor side effect of making `alt-enter` do nothing - as far as I can see - instead of just sending the `return` sequence, so we should state that.

I'll add something like the following:

> iTerm2, a popular terminal choice for Mac, must also be configured to enable this shortcut:
Go to *Preferences->Profiles->Default->Keys* and add a mapping that maps `⌥Return↩` to `Send Hex Codes: 0x1b 0x0d`. This can be done by clicking the + sign at the bottom to add a mapping, clicking the "Click to Set" area, pressing the desired key combination (⌥Enter a.k.a ⌥Return), choosing the "Send Hex Code" option from the drop-down menu and inserting the following string there: "0x1b 0x0d".

> Note that this will change the behavior of `alt+enter` for all terminal windows, and it will no longer send the `return` sequence.

**Comment by shaypal5** [comment]
Cool! Glad I was of a little help. :)
